.github/workflows/automation.yml

```diff
@@ -0,0 +1,133 @@
+
+      - name: Setup Java version
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'zulu'
+
+      - name: Install xmlstarlet
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y xmlstarlet
+
+      - name: Detect changed directories
+        id: changes
+        run: |
+          changed_files=$(git diff --name-only "${{ github.event.pull_request.base.sha }}" "${{ github.event.pull_request.head.sha }}")
+          echo "Changed files : $changed_files"
+          changed_dirs=$(dirname $changed_files | sort -u)
+          echo "Changed dirs : $changed_dirs"
+          root_dirs=()
+          for dir in $changed_dirs; do
+            if [[ $dir != *"."* ]]; then
+              relative_dir="$dir"
+              echo "Relative : $relative_dir"
+              while [[ ! -f "$relative_dir/pom.xml" ]]; do
+                relative_dir=$(dirname "$relative_dir")
+                echo "-- relative : $relative_dir"
+              done
+              if [[ -f "$relative_dir/pom.xml" ]]; then
+                root_dirs+=("$relative_dir")
+              fi
+            fi
+          done
+          echo "::set-output name=directories::${root_dirs[@]}"
+
+      - name: Display changed directories
+        run: |
+          echo "Changed directories: ${{ steps.changes.outputs.directories }}"
+
+      - name: Build and commit changes
+        run: |
+          for dir in ${{ steps.changes.outputs.directories }}; do
+            echo "Setting Variables $dir"
+            cd $dir
+            version=$(xmlstarlet sel -t -v "/*[local-name()='project']/*[local-name()='version']" -n pom.xml)
+            version_dir=build/$version
+            archival="$dir"_source.zip
+            echo "Building $dir With latest version : $version"
+            mvn clean install
+            echo "Archiving the build files to $archival"
+            zip -r $archival . -x "build/*"
+            echo "Creating an directory for version : $version_dir"
+            sudo mkdir -p $version_dir
+            echo "Copy build files to the location : $version_dir "
+            sudo mv $archival $version_dir/
+            sudo cp target/$dir.jar target/original-$dir.jar $version_dir/
+            ls $version_dir
+            echo "Committing the changes"
+            echo "Github Actor : ${{ github.actor }}"
+            git config --global user.email "${{ github.actor }}@users.noreply.github.com"
+            git config --global user.name "${{ github.actor }}"
+            git branch -a   
+            git checkout $GIT_BRANCH
+            git add .
+            git commit -m "build files added to the path $version_dir"
+            git push origin $GIT_BRANCH
+            cd $version_dir
+            echo "Updating the $dir addon in stacks"
+            curl --location --request PUT '${{ env.DEV_ADDON_URL }}/api/v1/plugins' \
+            --header 'Authorization: Bearer ${{ secrets.DEV_V1_TOKEN }}' \
```

Comment:
is DEV_V1_TOKEN same for all envs?

Reply:
@vikram-chaitanya No, we need add different based on the user in each stack.